simplifier/visitor/to_normal_form_visitor.py

```diff
@@ -48,8 +48,9 @@ def _recursive_call_visitor(self, operation: Union[BitwiseAnd, BitwiseOr]):
         For DNF-form, we do this if the operation is an OR-operation, because then we only have to bring the operands into DNF form.
         """
         if simplified := operation.simplify(keep_form=True):
-            self.visit(simplified)
-            return
+            if not simplified.world.compare(simplified, operation):
+                self.visit(simplified)
+                return
```

Comment:
Would it be possible to change it to the following:
```python
simplified = operation.simplify(keep_form=True)
if simplified.SYMBOL != operation.SYMBOL:
    self.visit(simplified)
    return
```
The compare function takes linear time with somehow increases the runtime. The only reason for this check is to guarantee that we have the same operation after the simplification. I am not happy with the SYMBOL, but I am not sure how to check that the simplified operation and the given operation have the same type.
To be sure that it is still correct, we should use `simplified` instead of `operation` in the following.

Reply:
Change has been removed as it is not needed anymore.